File: frontend/src/lib/queries/projects.ts
'use client'

import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
import { useApi } from '@/lib/api/use-api'

export type Project = {
  id: string
  name: string
  slug: string
  data_source_mode: 'files' | 'database' | 'google_sheet'
  status: string
  created_at: string
  updated_at: string
}

export function useProjects() {
  const { call, ready } = useApi()
  return useQuery({
    queryKey: ['projects'],
    queryFn: () => call<Project[]>('/api/v1/projects'),
    enabled: ready,
  })
}

export function useCreateProject() {
  const { call } = useApi()
  const queryClient = useQueryClient()
  return useMutation({
    mutationFn: (input: { name: string; data_source_mode?: Project['data_source_mode'] }) =>
      call<Project>('/api/v1/projects', { method: 'POST', json: input }),
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['projects'] })
    },
  })
}

export function useDeleteProject() {
  const { call } = useApi()
  const queryClient = useQueryClient()
  return useMutation({
    mutationFn: (projectId: string) =>
      call<void>(`/api/v1/projects/${projectId}`, { method: 'DELETE' }),
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['projects'] })
    },
  })
}

File: backend/app/schemas/projects.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class ProjectCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    data_source_mode: str = Field(default="files", pattern="^(files|database|google_sheet)$")


class ProjectOut(BaseModel):
    id: str
    name: str
    slug: str
    data_source_mode: str
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
